soroban-sdk/src/events.rs

```diff
@@ -58,13 +58,30 @@ impl Debug for Events {
     }
 }
 
+#[cfg(any(test, feature = "testutils"))]
+use crate::{testutils, xdr, Address, TryFromVal, TryIntoVal};
+
 pub trait Event {
     fn topics(&self, env: &Env) -> Vec<Val>;
     fn data(&self, env: &Env) -> Val;
 
     fn publish(&self, env: &Env) {
         env.events().publish_event(self);
     }
+
+    /// Convert this event and the given contract_id into a [`xdr::ContractEvent`] object.
+    #[cfg(any(test, feature = "testutils"))]
+    fn to_contract_event(&self, env: &Env, contract_id: &crate::Address) -> xdr::ContractEvent {
```

Comment:
I think we should drop the `env` and get the `env` out of the `Address`.

```suggestion
    fn to_contract_event(&self, contract_id: &crate::Address) -> xdr::ContractEvent {
```